### ISSUE TITLE
Fc after the update

### ISSUE THREAD

**Issue by rex07** [ram-on/SkyTube#475]
08-01 19:28:27.880 E/m.skytube.extr(15140): Not starting debugger since process cannot load the jdwp agent.
08-01 19:28:33.327 E/AndroidRuntime(15140): Process: free.rm.skytube.extra, PID: 15140
08-01 19:28:33.327 E/AndroidRuntime(15140): java.lang.RuntimeException: Unable to start activity ComponentInfo{free.rm.skytube.extra/free.rm.skytube.gui.activities.MainActivity}: java.lang.RuntimeException: com.google.android.gms.dynamite.DynamiteModule$LoadingException: No acceptable module found. Local version is 0 and remote version is 0.
08-01 19:28:33.327 E/AndroidRuntime(15140): 	at free.rm.skytube.gui.activities.BaseActivity.onCreate(BaseActivity.java:122)
08-01 19:28:33.327 E/AndroidRuntime(15140): 	at free.rm.skytube.gui.activities.MainActivity.onCreate(MainActivity.java:106)

**Comment by gzsombor** [comment]
What happened?

It seems to be coming from the Chromecast [code](https://github.com/ram-on/SkyTube/blob/a7a82e8fcbc7005a2123f4cb5b13f6957e66214b/app/src/extra/java/free/rm/skytube/gui/activities/BaseActivity.java#L122), but I can't reproduce it. @atomjack any idea, what could be the problem?

**Comment by atomjack** [comment]
Hmm I did find [this](https://stackoverflow.com/questions/38251934/no-acceptable-module-found-on-android-emulators-using-com-google-android-gms-p). @rex07, do you see anything like this in your logcat?

```W/GooglePlayServicesUtil: Google Play services out of date.  Requires 9256000 but found 9080030```

Are you running it on a device or in an emulator?

**Comment by rex07** [comment]
My device has no g apps installed maybe this is why the app fc

**Comment by atomjack** [comment]
@rex07 Yeah, that's why it's crashing, you don't have google play services installed. I've never tested it on a device without gapps, so it got by me. Chromecast support requires google play services, although obviously it shouldn't crash if you don't have that installed. 

@rex07 can you try [this apk](https://drive.google.com/open?id=13y-9q_Cbb8mFf0_YtHGk7CqO0DjC3jo6), and let me know if it still crashes? 

@gzsombor I've added some code to see if Google Play Services is available, and only if it is does it do all the setup for Chromecast support. I'm wondering if maybe something should be displayed when it doesn't find GPS, to let the user know that Chromecast support won't work without it.

@rex07 Is there a reason you're using the extra variant over the OSS one? Just the (proprietary) youtube video player?

**Comment by gzsombor** [comment]
Maybe, you can add a new preference to explicitly enable Chromecast support, and there could be a more elaborate check if everything is available, and shows a warning, if something is missing. Maybe the same could happen on app startup, when this is enabled, and if something is missing, then re-disable it, with similar warning.
 Or maybe, I'm overcomplicate this :smile:

**Comment by atomjack** [comment]
Yeah, I think that overcomplicates it. I think it could work like this:

If GPS is not available, display a popup explaining that Chromecast support requires it. Once the popup is shown, a flag will be saved in shared preferences to prevents the popup from ever showing again.

After that, tt will continue to check if GPS is available when the app starts/resumes, it will just do it silently, so if a user runs the app, sees the popup, and then sometime later installs gapps/GPS, then runs the app again, Chromecast support should work. 

I'm going to try to create a virtual device to test this on that doesn't have google play services on it and will report back.

What do you think, @gzsombor ?

**Comment by gzsombor** [comment]
Sure, that's much simpler :)

**Comment by atomjack** [comment]
@gzsombor Can you take a look at this branch I made, and let me know what you think? If you do have GPS installed, I'd like to have @rex07 try it out to make sure everything works properly, though (I will provide an updated apk, with the popup).

**Comment by gzsombor** [comment]
Maybe, you don't need to configure the dialog with all the default colors, but other than, this could work in my opinion.

**Comment by rex07** [comment]
@atomjack the app launch but it took a long time  

08-04 00:24:23.178 I/ActivityManager(845): Displayed free.rm.skytube.extra/free.rm.skytube.gui.activities.MainActivity: +12s525ms 

And if i click on any thing it fc without any error log


The reason why i use the extra
I'm the translator of this app to arabic language and i need to test if every thing is ok before the app go to the fdroid

**Comment by atomjack** [comment]
@rex07 Thanks for the input. Not sure why it would take longer for the app to come up. 

I've fixed the fc when you click on anything or minimize the app. Can you please download the apk again (I've updated it) and try it out?

@gzsombor I've updated the dialog to use better colors (white on darker background instead of black on white background) and updated that branch.

**Comment by Tester9920** [comment]
Hello there, I tested the version 2.962 but it crashed before starting up on my phone.
This app has been great for users of older version of Android and fix i
t please...

The information is below:

2019-08-07 06:34:50,ERROR,dalvikvm,Unknown,Could not find class 'android.app.NotificationChannel', referenced from method free.rm.skytube.app.SkyTubeApp.initChannels

**Comment by gzsombor** [comment]
@Tester9920 , this is a bit suprising, as there were no changes related to the SkyTubeApp.initChannels code, and there is still a check to the API level, and only call the NotificationChannel, if Android is recent enough.

**Comment by Tester9920** [comment]
@gizsombor, thank you very much for your reply.

I tried 2.95 the previous version and still found exactly the same log, yet the program works fine without problem.

So it might not be the main cause of the problem, but some issues still remain that makes unable to run it (at least) on my phone.

Sorry for the mistake and I really wish the problem is solved in the future and someone could help since I'm no expert of technical stuff...

**Comment by atomjack** [comment]
@Tester9920 Did you try the apk I posted above, to see if Skytube Extra still crashes? And, I'm assuming you don't have Google Play Services installed on your device, right?

**Comment by Tester9920** [comment]
@atomjack I tried the apk now, and it worked and didn't crash on my phone.
Google Play Services is actually installed, but the version is quite old (it says it's installed in 2014) and there are other apps that it needs to be updated as well.

I think it would be better to add an option to remind to not show "Google Play services not installed" dialog again for better browsing experience.

Thanks for great support, atomjack!

**Comment by atomjack** [comment]
@Tester9920 Awesome. I'll send a PR up for @gzsombor 

It already sets a flag that it showed that popup, and won't show the popup ever again once the flag is set. If you force stop the app and re-run it, it shouldn't show again.

**Comment by Tester9920** [comment]
@atomjack thank you so much, you're wonderful and you have saved my favorite SkyTube app! :)

**Comment by OMTDesign** [comment]
Hello

I have updated from Skytube 2.95 to 2.961 via Fdroid, and the application now crashes on start up. I downloaded Skytube Extra 2.962 and that had the same problem of crashing on start up. I tried to downgrade back to 2.95 via F droid, but there seems to be some internal error where file is not found in the repo. I managed to find and install Skytube Extra 2.95 and that is working fine for now.

I am using a Galaxy Note 4
Android 6.0.1
Kernel 3.10.9

I could not log a crash dump as I do not have the android SDK installed on my computer. I was not trying to support Chromecast on my device.

Thanks

**Comment by OMTDesign** [comment]
I tried to install the modified 2.961-test.apk that @atomjack posted but that fails to install.

**Comment by atomjack** [comment]
@OMTDesign You might try backing up your subs/bookmarks in the app (in Preferences), then uninstall the app, and then try installing my apk. My apk has a different signature key so it will have to be a clean install and not an upgrade (although it should usually just uninstall the other version for you when you try to install mine - more details on what happened when you tried to install would be helpful).

I assume you are using SkyTube Extra, and you aren't running the latest version of Google Play Services on your device? (My apk has the fix for that)

**Comment by OMTDesign** [comment]
@atomjack 
Thanks for making the modified application. I managed to get it installed, but the constant pop up telling me that GPlay services was not installed was too annoying for me, so I went back to using 2.95. Also there is a weird problem where a white bar would appear at the bottom of the screen when the video tried to maximise after the top status bar fades away. I think that is a problem of the new version of the app and not related to your edits.

![Screenshot_2019-08-16-09-41-49](https://user-images.githubusercontent.com/39666543/63129425-99dbc180-c00b-11e9-959b-4f3faace8da5.png)

**Comment by atomjack** [comment]
@OMTDesign Yeah, sorry about that, I fixed the popup thing after I posted the apk. The next release will have these fixes.

**Comment by atomjack** [comment]
@rex07 @OMTDesign, @gzsombor just released a new version of the app which includes my fixes, can you please let us know if that fixes it for you?

**Comment by JmD68** [comment]
I get the message app not installed.

**Comment by OMTDesign** [comment]
I tried the new apk (V 2.96.3) and you seemed to have fixed the issue with the pop up dialogue box.

But now there seems to be a host of new problems with Skytube. The Featured, Most Popular and Feed fail to load any videos. The videos that I have bookmarked work, but it fails to download channel information, and the search does not work at all. The issue with the white bar is still present and I found by turning off Immersive mode in the settings stops/fixes the issue.
